automated_packaging/common_functions.pm

```diff
@@ -0,0 +1,165 @@
+    $new_version=@_[0];
+    $old_version_non_regex=@_[1];
+    $file_name=@_[2];
+
+    # Update configuration file
+    open(CONFIGURATION_FILE, "<$file_name") || die "$file_name not found";
+    my @lines = <CONFIGURATION_FILE>;
+    close(CONFIGURATION_FILE);
+
+    my @newlines;
+    foreach(@lines) {
+       $_ =~ s/$old_version_non_regex/$new_version/g;
+       push(@newlines,$_);
+    }
+
+    open(CONFIGURATION_FILE, ">$file_name") || die "$file_name not found";
+    print CONFIGURATION_FILE @newlines;
+    close(CONFIGURATION_FILE);
+}
+
+# Get sorted PRs according to merged-at by collecting PRs up to given data (using created info)
+sub get_sorted_prs {
+    my $last_date = @_[0];
+    my $repo_name = @_[1];
+    my %sorted_pr_hash = ();
+    my $github_token = get_and_verify_token();
+
+    my $page_number=1;
+    do {
+        my $prs_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" 'https://api.github.com/repos/citusdata/$repo_name/pulls?base=master&state=all&page=$page_number'`;
+        my @prs = @{decode_json($prs_text)};
+
+        foreach my $pr ( @prs ) {
+            my %pr_hash = %$pr;
+
+            if (defined($pr_hash{'merged_at'})) {
+
+                $merged_date = substr($pr_hash{'merged_at'}, 0 ,10);
+
+                if ($merged_date lt $last_date) {
+                    last;
+                }
+
+                $sorted_pr_hash{$pr_hash{'merged_at'}} = $pr_hash{'url'};
+            }
+        }
+
+        $page_number += 1;
+    } while($merged_date gt $last_date);
+
+    my @keys = reverse sort keys(%sorted_pr_hash);
+    my @vals = @sorted_pr_hash{@keys};
+
+    print("PRs has been read in the merge order ..." . "\n");
+
+    return @vals;
+}
+
+# Search for the backport label given a label array
+sub has_backport_label {
+
+    my @labels = @{$_[0]};
+
+    foreach my $label (@labels) {
+        if (ref $label eq ref {}) {
+            %label_hash=%{$label};
+            if ($label_hash{'name'} eq 'backport') {
+                return 1;
+            } 
+        }   
+    }
+
+    return 0;
+}
+
+# Creates changelog entries up to the given last date. It extracts lines starting with DESCRIPTION: from 
+# PR messages to generate these entries. You can use different repo names to generate changelog entries for
+# different repos.
+sub create_release_changelog {
+    my $last_date = @_[0];
+    my $repo_name = @_[1];
+    my $is_point_release = @_[2];
+
+    my @comment_lines = ();
+    my $github_token = get_and_verify_token();
+
+    my @sorted_pr_urls = get_sorted_prs($last_date, $repo_name);
+
+    foreach my $pr_url ( @sorted_pr_urls ) {
+        print("Getting information for " . $pr_url . "\n");
+        my $pr_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" '$pr_url'`;
+        my %pr_hash = %{decode_json($pr_text)};
+        my $add_to_changelog = 1;
+
+        if ($is_point_release == 1) {
```

Comment:
I am not entirely sure but I guess we do not need ` == 1` part.

If this is the case, there are some other instances of ` == 1` . Please remove them too.